rabbit-escape-render/src/rabbitescape/render/gameloop/GeneralPhysics.java

```diff
@@ -104,6 +124,25 @@ public boolean gameRunning()
         return ( world.completionState() == World.CompletionState.RUNNING );
     }
 
+    /**
+     * Take actions for demo mode, eg drop tokens.
+     */
+    private void doInterpreterActions()
+    {
+        SolutionTimeStep stp = solutionInterpreter.next( CompletionState.RUNNING );
+        for ( TimeStepAction action: stp.actions )
+        {
+            if ( action instanceof SelectAction )
+            {
+                actionTypeSwitch.caseSelectAction( (SelectAction)action );
+            }
+            else if ( action instanceof PlaceTokenAction )
+            {
+                actionTypeSwitch.casePlaceTokenAction( (PlaceTokenAction)action );
+            }
```

Comment:
Yeah I managed to figure out that you need an interface to reverse that dependency but I was suggesting making onespecifically rather than reusing TimeStepActionTypeSwitch. Those type switch interfaces were intended to be used like switch statements so I was surprised to see one used like s normal interface. That doesn't mean it's wrong but the fact that not all the interface methods are used is maybe a bit suspicious.